src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -983,16 +984,26 @@ public void refreshStudyInformation() {
             String entitySetName,
             String token ) {
         try {
-
             ApiClient apiClient = new ApiClient( () -> token );
             EntitySetsApi entitySetsApi = apiClient.getEntitySetsApi();
             SearchApi searchApi = apiClient.getSearchApi();
+            EdmApi edmApi = apiClient.getEdmApi();
 
             String participantEntitySetName = ChronicleServerUtil.getParticipantEntitySetName( studyId );
             UUID participantEntitySetId = entitySetsApi.getEntitySetId( participantEntitySetName );
-
             UUID srcEntitySetId = entitySetsApi.getEntitySetId( entitySetName );
 
+            // create a dictionary from property fqn to title of
+            EntityType appDataET = edmApi.getEntityType( entitySetsApi.getEntitySet( srcEntitySetId ).getEntityTypeId() );
+            Map<String, String> columnNameDictionary = appDataET.getProperties()
+                    .stream()
+                    .collect(
+                            Collectors.toMap(
+                                    k -> edmApi.getPropertyType( k ).getType().getFullQualifiedNameAsString(),
+                                    k -> entitySetsApi.getEntitySetPropertyMetadata( srcEntitySetId, k ).getTitle()
+                            )
+                    );
```

Comment:
There's 12 properties for this entity type, which means this is making 24 API requests to gather all this data. This doesn't seem ideal.

Reply:
Done, only 2 requests now.